File: go-backend/internal/model/entity/request_log.go
package entity

import "time"

type RequestLog struct {
	ID               int64     `json:"id,string" gorm:"column:id;primaryKey;autoIncrement"`
	TraceID          string    `json:"traceId" gorm:"column:traceId"`
	UserID           *int64    `json:"userId,string,omitempty" gorm:"column:userId"`
	APIKeyID         *int64    `json:"apiKeyId,string,omitempty" gorm:"column:apiKeyId"`
	ModelID          *int64    `json:"modelId,string,omitempty" gorm:"column:modelId"`
	RequestModel     string    `json:"requestModel" gorm:"column:requestModel"`
	ModelName        string    `json:"modelName" gorm:"column:modelName"`
	RequestType      string    `json:"requestType" gorm:"column:requestType"`
	Source           string    `json:"source" gorm:"column:source"`
	PromptTokens     int       `json:"promptTokens" gorm:"column:promptTokens"`
	CompletionTokens int       `json:"completionTokens" gorm:"column:completionTokens"`
	TotalTokens      int       `json:"totalTokens" gorm:"column:totalTokens"`
	Cost             float64   `json:"cost" gorm:"column:cost"`
	Duration         int       `json:"duration" gorm:"column:duration"`
	Status           string    `json:"status" gorm:"column:status"`
	ErrorMessage     string    `json:"errorMessage" gorm:"column:errorMessage"`
	ErrorCode        string    `json:"errorCode" gorm:"column:errorCode"`
	RoutingStrategy  string    `json:"routingStrategy" gorm:"column:routingStrategy"`
	IsFallback       int       `json:"isFallback" gorm:"column:isFallback"`
	ClientIP         string    `json:"clientIp" gorm:"column:clientIp"`
	UserAgent        string    `json:"userAgent" gorm:"column:userAgent"`
	CreateTime       time.Time `json:"createTime" gorm:"column:createTime"`
	UpdateTime       time.Time `json:"updateTime" gorm:"column:updateTime"`
}

func (RequestLog) TableName() string {
	return "request_log"
}

File: python-backend/app/models/request_log.py
"""Request log model."""

from __future__ import annotations

from datetime import datetime

from sqlalchemy import BigInteger, DateTime, Index, Integer, String, Text, func
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base


class RequestLog(Base):
    __tablename__ = "request_log"
    __table_args__ = (
        Index("idx_userId", "userId"),
        Index("idx_apiKeyId", "apiKeyId"),
        Index("idx_createTime", "createTime"),
    )

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True, autoincrement=True)
    trace_id: Mapped[str | None] = mapped_column("traceId", String(64), nullable=True)
    user_id: Mapped[int | None] = mapped_column("userId", BigInteger, nullable=True)
    api_key_id: Mapped[int | None] = mapped_column("apiKeyId", BigInteger, nullable=True)
    model_id: Mapped[int | None] = mapped_column("modelId", BigInteger, nullable=True)
    request_model: Mapped[str | None] = mapped_column("requestModel", String(128), nullable=True)
    model_name: Mapped[str] = mapped_column("modelName", String(128), nullable=False)
    request_type: Mapped[str] = mapped_column("requestType", String(32), nullable=False, default="chat")
    source: Mapped[str] = mapped_column(String(32), nullable=False, default="web")
    prompt_tokens: Mapped[int] = mapped_column("promptTokens", Integer, nullable=False, default=0)
    completion_tokens: Mapped[int] = mapped_column("completionTokens", Integer, nullable=False, default=0)
    total_tokens: Mapped[int] = mapped_column("totalTokens", Integer, nullable=False, default=0)
    duration: Mapped[int] = mapped_column(Integer, nullable=False, default=0)
    status: Mapped[str] = mapped_column(String(32), nullable=False, default="success")
    error_message: Mapped[str | None] = mapped_column("errorMessage", Text, nullable=True)
    error_code: Mapped[str | None] = mapped_column("errorCode", String(64), nullable=True)
    routing_strategy: Mapped[str | None] = mapped_column("routingStrategy", String(32), nullable=True)
    is_fallback: Mapped[int] = mapped_column("isFallback", Integer, nullable=False, default=0)
    client_ip: Mapped[str | None] = mapped_column("clientIp", String(64), nullable=True)
    user_agent: Mapped[str | None] = mapped_column("userAgent", String(512), nullable=True)
    create_time: Mapped[datetime] = mapped_column(
        "createTime",
        DateTime,
        nullable=False,
        default=datetime.utcnow,
        server_default=func.current_timestamp(),
    )
    update_time: Mapped[datetime] = mapped_column(
        "updateTime",
        DateTime,
        nullable=False,
        default=datetime.utcnow,
        onupdate=datetime.utcnow,
        server_default=func.current_timestamp(),
    )
